airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalAttemptExecution.java

```diff
@@ -46,7 +47,7 @@
   private final Consumer<Path> mdcSetter;
   private final CancellationHandler cancellationHandler;
   private final Supplier<String> workflowIdProvider;
-  private final JobPersistence jobPersistence;
+  private final AirbyteApiClient airbyteApiClient;
```

Comment:
Is there an argument for wrapping the client and passing in a paired down interface? Right now I read this and the worker could be calling _any_ endpoint in the config api. I think we only ever want it to call 2 of them (or so). Would it be more expressive to create the `WorkerScopedApiClient` or something. Might also be easier to mock.

Reply:
+1 on this. Not sure if we want to act on this in this PR though. I think @jdpgrailsdev had a PR to break down the API into multi-class, I'd expect we should do the same for the client libraries.

Reply:
Continued in child PR here: https://github.com/airbytehq/airbyte/pull/15902/files#r953198331